### PR TITLE
[pkg/stanza] Windows input operator logs noisy error while collecting successfully

### PR DESCRIPTION
**Pull request by kuiperda**
**Description:** <Describe what has changed.>

If `Input::read` returns 0, the operator will attempt to open the subscription again. This happens after a resubscribe *or* whenever there are no event logs to read, which is often. As a result, this error gets logged constantly:
```
"Failed to re-open subscription for remote server"..."error: subscription handle is already open"
```
To fix this, the handle should only attempt to re-open when the resubscribe happens. 

This is a followup to  #35175 which fixed that resubscribe issue for the user that raised it, but they have also been confused by this error being logged. 

**Link to tracking Issue:** <Issue number if applicable> **N/A**

**Testing:** <Describe what testing was performed and which tests were added.> Manually verified log collection is successful before and after remote host reconnects, and that the noisy error is silenced except during resubscribe, where it is relevant.

**Documentation:** <Describe the documentation added.> **N/A**

### CODE REVIEW

**Review by djaglowski — COMMENTED** [review]
LGTM. Please take a look @pjanotti

File: pkg/stanza/operator/input/windows/input.go
Comment (pjanotti) [inline pkg/stanza/operator/input/windows/input.go:187]: Double-check me here, but, I think that this will be the only call to `read` and we don't care about the int return. It should be removed from the function signature.

File: pkg/stanza/operator/input/windows/input.go
Comment (pjanotti) [inline pkg/stanza/operator/input/windows/input.go:211]: Just making clear that this error will be logged until the remote server is reachable again.

**Review by pjanotti — COMMENTED** [review]
Not from this change, but, this seems to leak the remote session handle, since closing the subscription doesn't closes it. That said I think this happened because the way the code is structured: this would have been avoided if the remote session handle was also part of the subscription.

File: pkg/stanza/operator/input/windows/input.go
Comment (pjanotti) [reply pkg/stanza/operator/input/windows/input.go:187]: @kuiperda check this please, after that we should be good to merge, we can work on refactoring the subscription in a follow-up.

Comment (kuiperda) [reply pkg/stanza/operator/input/windows/input.go:187]: Yes, good call.

File: pkg/stanza/operator/input/windows/input.go
Comment (kuiperda) [reply pkg/stanza/operator/input/windows/input.go:211]: Yes. I think that is fine, though a backoff loop may be appropriate if the host is down for longer than a few seconds.

**Comment by kuiperda** [comment]
@pjanotti Thanks for your feedback. I updated the return for that function.

**Review by pjanotti — APPROVED** [review]
@kuiperda LGTM - if you can't start working soon on the remote session handle leak, could you, please, create a issue so we don't forget about it?

**Comment by kuiperda** [comment]
@pjanotti I'll make the issue but likely will not have time to work on it.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**
